src/database/bplus/mod.rs

```diff
@@ -0,0 +1,102 @@
+//! # Bplus Database [WIP]
+//!
+//! This module provides a database implementation backed by a
+//! persistent bplus tree.
+//!
+//! #### Authorship
+//!
+//! - Benjamin Riley Zimmerman, 9/11/2024 (bz931@berkeley.edu)
+
+/* IMPORTS */
+
+use std::path::Path;
+
+use anyhow::Result;
+use bitvec::prelude::{BitSlice, Msb0};
+
+#[allow(unused_imports)]
+use crate::{
+    model::State,
+    database::{Persistence, KVStore, Persistent, Tabular, Record, Schema, SchemaBuilder},
+};
+
+/* CONSTANTS */
+
+
+
+/* DATABASE DEFINITION */
+
+#[allow(dead_code)]
+pub struct Database<'a> {
+    data: &'a str,
```

Comment:
I assume this will represent the in-memory cache:

1) we'll need it to be `mut`
2) it's better to treat is as a `[u8]` (contiguous sequence of bytes)

```
data: &'a mut [u8]
```

Reply:
ah no this was to avoid build error of unused lifetime. i tried using `#[allow(unused)]` but couldnt fix build error, so i just used a tmp string

Reply:
gotcha sg